### PR TITLE
Generate VethName based on podname and namespace in CNI

### PR DESCRIPTION
**Pull request by tamilmani1989**
**What this PR does / why we need it**:
Generate VethName based on podname and namespace in CNI. The function will generate hash of the key provided and return first 11 characters of hash. The key is "podname.podnamespace" in CNI.  

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: cni/network/network.go [hunk 1]
```diff
@@ -168,6 +169,12 @@ func (plugin *netPlugin) Add(args *cniSkel.CmdArgs) error {
 		return err
 	}
 
+	k8sPodName := string(podCfg.K8S_POD_NAME)
+	if len(k8sPodName) == 0 {
+		err = plugin.Errorf("No k8s pod namespace provided.")
+		return err
```
Comment (jaer-tsun) [inline cni/network/network.go]: just return plugin.Errorf(...)?

File: network/api.go [hunk 2]
```diff
@@ -17,4 +17,6 @@ var (
 	errEndpointNotFound   = fmt.Errorf("Endpoint not found")
 	errEndpointInUse      = fmt.Errorf("Endpoint is already joined to a sandbox")
 	errEndpointNotInUse   = fmt.Errorf("Endpoint is not joined to a sandbox")
+
+	OptVethName = "vethname"
```
Comment (jaer-tsun) [inline network/api.go]: is there a const file to add this to?

File: network/endpoint_linux.go [hunk 3]
```diff
@@ -19,28 +21,43 @@ const (
...
+	hostVEthInterfacePrefix = commonInterfacePrefix + "v"
 
 	// Prefix for container network interface names.
 	containerInterfacePrefix = "eth"
 )
 
+func generateVethName(key string) string {
+	h := sha1.New()
+	h.Write([]byte(key))
+	return hex.EncodeToString(h.Sum(nil))[:11]
```
Comment (jaer-tsun) [inline network/endpoint_linux.go]: return hex.EncodeToString( sha1.Sum( []byte(key) ) )[:11]

File: network/api.go (hunk 2 above)
Comment (tamilmani1989) [reply network/api.go]: we are following the same thing in ipam/api.go

**Pull request merged**